extensions/amp-story/0.1/progress-bar.js

```diff
@@ -59,6 +60,9 @@ export class ProgressBar {
 
     /** @private @const {!../../../src/service/vsync-impl.Vsync} */
     this.vsync_ = Services.vsyncFor(this.win_);
+
+    /** @private {!Object<string, number>} */
+    this.pageIdMap_ = map();
```

Comment:
This doesn't seem like it belongs in `progress-bar`.  `progress-bar` is UI code that shouldn't really have an understanding of pages or their requirements (e.g. the assumption that pages have IDs).  Ideally the code as at was before wouldn't even refer to "page" count or "page" index, but more like "segment" count.

Feels like this could go in amp-story?  There may also be other index-based operations in stories anyway, which would then need this same calculation if something gets inserted.